src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs

```diff
@@ -893,11 +894,23 @@ public string RoleClaimType
         /// </summary>
         public IEnumerable<string> ValidAudiences { get; set; }
 
+        private string _validIssuer;
+
         /// <summary>
         /// Gets or sets a <see cref="string"/> that represents a valid issuer that will be used to check against the token's issuer.
         /// The default is <c>null</c>.
         /// </summary>
-        public string ValidIssuer { get; set; }
+        public string ValidIssuer
```

Comment:
Span<T> including ReadOnlySpan<T> can only be allocated on a stack. This prevents them from being a member of a class as that requires a heap allocation. Consider introducing a ReadOnlyMemory<T> to store the UTF8 representation of an issuer. 

ReadOnlyMemort<T>.AsSpan().Equals(..) can then be used during issuer validation.